tenzir/tests/run.py

```diff
@@ -501,13 +507,105 @@ def run(self, test: Path, update: bool) -> bool:
         return run_simple_test(test, update=update, ext="txt")
 
 
+class ConfigRunner(Runner):
+    def __init__(self):
+        super().__init__(prefix="config")
+
+    def collect_tests(self):
+        return self.collect_with_ext(".tql")
+
+    def purge(self):
+        pass
+
+    def run(self, test: Path, update: bool) -> bool:
+        test_config = parse_test_config(test)
+        expect_failure = test_config.get("error", False)
+
+        # Try to start a node with the given configuration
+        env, config_args = get_test_env_and_config_args(test)
+        with tempfile.TemporaryDirectory() as temp_dir:
+            try:
+                node_cmd = [
+                    TENZIR_NODE_BINARY,
+                    "--bare-mode",
+                    "--console-verbosity=warning",
+                    "--console-format=%v",
+                    f"--state-directory={Path(temp_dir) / 'state'}",
+                    f"--cache-directory={Path(temp_dir) / 'cache'}",
+                    "--endpoint=localhost:0",
+                    "--print-endpoint",
+                    *[x for x in config_args if x is not None],
+                ]
+
+                # Capture both stdout and stderr for reference file
+                proc = subprocess.run(
+                    node_cmd,
+                    stdout=subprocess.PIPE,
+                    stderr=subprocess.PIPE,
+                    text=True,
+                    env=env,
+                    timeout=parse_test_config(test)["timeout"],
+                )
+
+                # Check if the process exited as expected
+                node_started = proc.returncode == 0 and "localhost:" in proc.stdout
+
+                if expect_failure and node_started:
+                    report_failure(
+                        test,
+                        f"└─▶ \033[31mNode started successfully but was expected to fail\033[0m",
+                    )
+                    return False
+                elif not expect_failure and not node_started:
+                    report_failure(
+                        test,
+                        f"└─▶ \033[31mNode failed to start but was expected to succeed\033[0m",
+                    )
+                    return False
+
+                # Create content for the reference file (combined stdout and stderr)
+                output = proc.stdout + "\n" + proc.stderr
```

Comment:
Ah, now I see where the newline is coming from. Not too much of a problem, but maybe we can also make it conditional. 

Only slightly related: You can also pass `subprocess.STDOUT` to `stderr` above I think, don't remember how exactly that differs, but if you want a separator that's not an option.

Reply:
Fwiw, this also reminds me of `--dump-diagnostics`, which writes the diagnostics to stdout and instead of stderr.